langstream-agents/langstream-agent-s3/src/main/java/ai/langstream/agents/s3/S3Processor.java

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ai.langstream.agents.s3;
+
+import ai.langstream.ai.agents.commons.JsonRecord;
+import ai.langstream.ai.agents.commons.MutableRecord;
+import ai.langstream.api.runner.code.AbstractAgentCode;
+import ai.langstream.api.runner.code.AgentProcessor;
+import ai.langstream.api.runner.code.Header;
+import ai.langstream.api.runner.code.Record;
+import ai.langstream.api.runner.code.RecordSink;
+import ai.langstream.api.runtime.ComponentType;
+import com.samskivert.mustache.Mustache;
+import com.samskivert.mustache.Template;
+import io.minio.GetObjectArgs;
+import io.minio.GetObjectResponse;
+import io.minio.MinioClient;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import lombok.AllArgsConstructor;
+import lombok.ToString;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class S3Processor extends AbstractAgentCode implements AgentProcessor {
+    private String bucketName;
+    private MinioClient minioClient;
+    private Template objectTemplate;
+
+    @Override
+    public void init(Map<String, Object> configuration) throws Exception {
+        bucketName = configuration.getOrDefault("bucketName", "langstream-source").toString();
+        String endpoint =
+                configuration
+                        .getOrDefault("endpoint", "http://minio-endpoint.-not-set:9090")
+                        .toString();
+        String username = configuration.getOrDefault("access-key", "minioadmin").toString();
+        String password = configuration.getOrDefault("secret-key", "minioadmin").toString();
+        String region = configuration.getOrDefault("region", "").toString();
+        String objectName = configuration.getOrDefault("objectName", "").toString();
```

Comment:
what about providing a sensible default, like "{{value.objectname}}" ?
ths way people don't have to bother with mustache for simple cases